### ISSUE TITLE
Mis-inplementation of JS divergence

### ISSUE THREAD

**Issue by greatwallet** [nv-tlabs/datasetGAN_release#27]
Hi, according to definition of JS divergence (as mentioned in your [supp file](https://drive.google.com/file/d/1td1nP8FP0axHXFxl9_EXCtHQHhnaUQl8/view)), JS divergence is calculated as the difference of 
entropy of average **probabilities** and average of entropies.

![image](https://user-images.githubusercontent.com/30019142/132695150-8a226211-5dbf-4828-ba19-279f715e9e7a.png)
 
However in your code, the first term of JS, aka the difference of entropy of average **probabilities** is implemented as:
https://github.com/nv-tlabs/datasetGAN_release/blob/dee6d7d2dfbdffdff27b1cd4433b41f2521e5ff1/datasetGAN/train_interpreter.py#L304

where `mean_seg` is defined as average segmentation map of 10 outputs of ensembled `pixel_classifier`s. 

Specifically, I have traced the implementation of `mean_seg` --> 
https://github.com/nv-tlabs/datasetGAN_release/blob/dee6d7d2dfbdffdff27b1cd4433b41f2521e5ff1/datasetGAN/train_interpreter.py#L302
--> 
https://github.com/nv-tlabs/datasetGAN_release/blob/dee6d7d2dfbdffdff27b1cd4433b41f2521e5ff1/datasetGAN/train_interpreter.py#L291-L294
--> `img_seg`
https://github.com/nv-tlabs/datasetGAN_release/blob/dee6d7d2dfbdffdff27b1cd4433b41f2521e5ff1/datasetGAN/train_interpreter.py#L282-L284

In fact, `img_seg` are all **unnormalized probabilities**, aka **logits** defined in pytorch distribution's argument. I think in the code you attempted to do average upon **logits** instead of **probabilies** (since you have commented out `Sigmoid` in `pixel_classifier`)
https://github.com/nv-tlabs/datasetGAN_release/blob/dee6d7d2dfbdffdff27b1cd4433b41f2521e5ff1/datasetGAN/train_interpreter.py#L68-L92

## TL; DR
The unlawful commutation of `softmax` and linear operation leads to mis-implementation of JS divergence.
![image](https://user-images.githubusercontent.com/30019142/132698300-688d9af6-703e-4092-91ad-093757e6a7d9.png)

**Comment by arieling** [comment]
Thank you a lot for pointing this out. We are looking into it.

**Comment by arieling** [comment]
@greatwallet Thank you again for pointing this bug out! 
We have fixed the bug in the commit of 
https://github.com/nv-tlabs/datasetGAN_release/commit/d9564d4d2f338eaad78132192b865b6cc1e26cac.
The number is also updated in README.
